llvm/lib/ProfileData/SampleProf.cpp

```diff
@@ -278,6 +279,110 @@ void FunctionSamples::findAllNames(DenseSet<FunctionId> &NameSet) const {
   }
 }
 
+namespace {
+// A class to keep a string invariant where it is appended with contents in a
+// recursive function.
+struct ScopedString {
+  std::string &String;
+
+  size_t Length;
+
+  ScopedString(std::string &String) : String(String), Length(String.length()) {}
+
+  ~ScopedString() {
+    assert(String.length() > Length);
+    String.resize(Length);
+  }
+};
+} // namespace
+
+// CanonicalName is invariant in this function. It should already contain the
+// canonical name of the current FunctionSamples.
+uint64_t FunctionSamples::eraseInlinedCallsites(const llvm::Regex &Re,
+                                                std::string &CanonicalName,
+                                                bool MatchCallTargets,
+                                                bool EraseMatch) {
+  uint64_t Result = 0;
+
+  ScopedString SaveString1(CanonicalName);
```

Comment:
I was confused why it created a variable but never used.. then I found it leveraged a dtor to resize string.. can we avoid this? is it possible just use a length or create a tmp string to do this?